src/sidebar/services/tags.js

```diff
@@ -41,15 +44,21 @@ export class TagsService {
     //   (e.g. tag "pink banana" matches query "ban"), OR
     // * tag has substring query occurring after a non-word character
     //   (e.g. tag "pink!banana" matches query "ban")
```

Comment:
Given the changes in this branch, I think it would be helpful to replace the multi-line comment above the `let regex...` line with one which describes the overall intent of the code in a language-agnostic way, rather than the specific matching criteria. For example:

```js
// Match any tag where the query is a prefix of the tag or a word within the tag.
```